### PR TITLE
fix(ui): fixed a memory leak and warning in topbar.cpp

### LINKED ISSUE #2746: memory leak in SetupTopBarWidgetsPage
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

1. the menu member variable of the SetupTopBArWidgetsPages gets initialised with itself, which is pointless
2. in deleteLater a new ScreenMenu is created with "new ScrenMenu(0);", but nowhere stored.

### Expected Behavior

no memory leak

### Steps To Reproduce

use the source

### Version

Nightly (Please give date/commit below)

### Transmitter

Other (Please specify below)

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by gagarinlg**
Fixes #2746

### CODE REVIEW

**Comment by pfeerick** [comment]
Just double checking whether the behavioural change was intentional or not... i.e. screen settings -> user interface (first tab) -> setup widgets (top bar) -> make some change or not -> either RTN/tap ETX logo top left, and it returns to the main view, rather than previous (screen settings) screen. Just seems inconsistent? 🤷

**Comment by gagarinlg** [comment]
It is not intentional and I have no idea why the changes trigger such a change

File: radio/src/gui/colorlcd/topbar.cpp [hunk 1]
```diff
@@ -83,8 +82,5 @@ void SetupTopBarWidgetsPage::deleteLater(bool detach, bool trash)
   // and continue async deletion...
   FormWindow::deleteLater(detach, trash);
 
-  // restore screen setting tab on top
-  new ScreenMenu(0);
-
```
Comment (pfeerick) [inline radio/src/gui/colorlcd/topbar.cpp:88]: This is the culprit - on closing the `SetupTopBarWidgetsPage` page the first screen setting page isn't being re-created... just like in `widgets_setup.cpp deleteLater`

What I don't know is if this was part of the memory leak you were chasing, or wrong place at the wrong time ;)

Comment (raphaelcoeffic) [reply radio/src/gui/colorlcd/topbar.cpp]: This is right: this is basically what sends the user back to the "User Interface" page, so it needs to stay.

**Comment by raphaelcoeffic** [comment]
Not a memory leak after all, but it is right that the `menu` attribute is superfluous and on top of that, initialised the wrong way (with itself).

**Review by raphaelcoeffic — APPROVED** [review]

**Comment by gagarinlg** [comment]
> Not a memory leak after all, but it is right that the `menu` attribute is superfluous and on top of that, initialised the wrong way (with itself).

OK, maybe I should not do such stuff while sitting in a train.
It still looks strange and maybe should get a comment, because it gets no parent in thus looks orphaned.

**Pull request merged**